exporter/signalfxexporter/translation/metricdata_to_signalfxv2.go

```diff
@@ -76,15 +76,30 @@ var (
 	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
-func metricDataToSignalFxV2(
+func MetricDataToSignalFxV2(
 	logger *zap.Logger,
+	metricTranslator *MetricTranslator,
 	md consumerdata.MetricsData,
-) (sfxDataPoints []*sfxpb.DataPoint, numDroppedTimeSeries int, err error) {
+) (sfxDataPoints []*sfxpb.DataPoint, numDroppedTimeSeries int) {
+	sfxDataPoints, numDroppedTimeSeries = metricDataToSfxDataPoints(logger, md)
+	if metricTranslator != nil {
+		metricTranslator.TranslateDataPoints(sfxDataPoints)
```

Comment:
do you think this translation and the sanitization could be moved to `metricDataToSfxDataPoints`? Seems like since we already iterate over the datapoints in that method, we can avoid going over it again? What do you think?

Reply:
I don't see any benefit from moving it to `metricDataToSfxDataPoints` to be honest. Additional iteration doesn't seem to expensive comparing to all the other logic, and I'd really like to separate out all the additional metric translations from the core data conversion logic. At some point, once backend supports all the Otel convention we should be able to remove these couple of lines, instead of going through the data conversion logic.

Reply:
I agree some of the translations we may want to do would fit better as a post processing step. But in this case I guess if we do these translations before converting `consumerdata.MetricsData` to `[]*sfxpb.DataPoint`, doing the mapping once per `consumerdata.MetricsData` is sufficient whereas later on we have to do it once per dimension per `sfxpb.DataPoint`. Thoughts?

Reply:
We cannot mutate `consumerdata.MetricsData` because it'll affect other exporters that user sets in the pipeline, so in that case would need to copy `consumerdata.MetricsData` objects before making any modifications. That's why I think it's better to do translation on final data structure that is already a copy and can be mutated.

Reply:
Also translation that can be done once on `consumerdata.MetricsData` is resource/node labels renaming, which is pretty small part comparing to all the other translations we have to do: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/476/files#diff-a964270a3bd3600cfd48aa1d7f5f856f

Reply:
I wasn't recommending we update `consumerdata.MetricsData` but rather do the translation alongside the `consumerdata.MetricsData` -> `[]*sfxpb.Datapoint` conversion. However, if you don't think we'll gain a lot in terms efficiency, I'm good with this approach.